app/commands/v2/put_link_set.rb

```diff
@@ -9,7 +9,29 @@ def call
           version.increment
           version.save! if link_set.valid?
 
-          link_set.links = merge_links(link_set.links, link_params.fetch(:links))
+          links_hash = link_params.fetch(:links)
+
+          links_hash.each do |link_type, target_content_ids|
+            if target_content_ids.empty?
+              Link.where(link_set: link_set, link_type: link_type).delete_all
+            else
+              old_target_ids = Link.where(link_type: link_type, link_set: link_set).map(&:target_content_id)
```

Comment:
You could use `link_set.links.where(link_type: link_type).pluck(: target_content_id)`. 

`pluck` avoids loading all records and is faster.